feat: Added Entity Framework Core's schemas

kyranet opened the pull request.

kyranet committed:
feat: Added (painful) EFC auto-generated schemas

They need clean-up, seriously.

kyranet committed:
cleanup: First step at cleaning everything up

kyranet committed:
build: Resolved build errors

Tylertron1998 commented on Skyra/Database/Models/User.cs:
```diff
@@ -0,0 +1,25 @@
+		public bool ModerationDm { get; set; } = true;
+		public DateTime? NextDaily { get; set; } = null;
+		public DateTime? NextReputation { get; set; } = null;
+		public int Vault { get; set; } = 0;
```
can the Vault be negative? If not, consider using an unsigned type - perhaps ushort?

kyranet replied:
Actually that should be `ulong`

Tylertron1998 commented on Skyra/Database/Models/Guild.cs:
```diff
@@ -0,0 +1,361 @@
+		None
+	}
+
+	public abstract class GuildModerationShared
```
perhaps consider GuildModerationBase as a name - seems to make more sense to me.

Tylertron1998 commented on Skyra/Database/Models/Guild.cs:
```diff
@@ -0,0 +1,361 @@
+
+	public struct GuildTag
+	{
+		[JsonProperty("name")]
```
are these JsonPropertyAttributes needed? AFAIK; at least with Newtonsoft.Json - the casing doesn't matter; `name` and `Name` is the same thing to it.

Tylertron1998 reviewed (commented):
Some questions in review; also 
- why are all of these classes? Can they not be structs? If so, they'd be unmodifiable in efcores cache.
- why does every property have a getter and a setter? Is this because of efcore requiring?

other then this; looks good to me. Good job and well done!

kyranet committed:
cleanup: Move entities to their own files for organization

kyranet commented:
- Classes are the way:tm: to make entities in EF, at least, according to [many] ([seriously]) [docs].
- Ditto ^

[many]: https://docs.microsoft.com/en-us/ef/core/modeling/
[seriously]: https://www.learnentityframeworkcore.com/
[docs]: https://www.entityframeworktutorial.net/code-first/simple-code-first-example.aspx

Tylertron1998 commented:
Looks good to me!

kyranet committed:
fix: Resolved complex type problem

Next problem:

System.InvalidOperationException: The child/dependent side could not be determined for the one-to-one relationship between 'GuildAutoModeration.Attachment' and 'GuildModerationAttachment.Guild'. To identify the child/dependent side of the relationship, configure the foreign key property. If these navigations should not be part of the same relationship configure them without specifying the inverse. See http://go.microsoft.com/fwlink/?LinkId=724062 for more details.

kyranet committed:
fix: Remove testing code

kyranet committed:
cleanup: Move classes to their respective files

Current error:

System.InvalidOperationException: The entity type 'GuildActionTrigger' requires a primary key to be defined. If you intended to use a keyless entity type call 'HasNoKey()'.

kyranet committed:
cleanup: Seal all model classes, refactor Guild.Actions

kyranet committed:
fix: Resolved the last bug, EFC is now working!

kyranet committed:
misc: Added lots of docs and fixed Roslyn suggestions

kyranet committed:
refactor: Revamp GuildRoles table

kyranet committed:
cleanup: Converted the last class to a struct

kyranet committed:
fix: Resolved all build warnings

kyranet committed:
cleanup: Optimize SQL tables, reduce the amount of indexes drastically

kyranet committed:
misc: Updated more names, turned GuildModerationSoftAction into a struct

kyranet committed:
feat: Added setters to GuildModerationSoftAction

kyranet committed:
refactor: Use flag enum instead

kyranet committed:
docs: Finished writing docs and entities
